apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx

```diff
@@ -378,11 +388,38 @@ export function ActivityDetails(props: {
           {/* Order Currency Logo */}
           {inputToken && outputToken && (
             <ActivityVisual>
-              <CurrencyLogoPair
-                sellToken={inputToken}
-                buyToken={isBridgeOrder ? swapAndBridgeOverview?.targetCurrency || outputToken : outputToken}
-                tokenSize={32}
-              />
+              {isBridgeOrder && order ? (
+                (() => {
+                  const isLocalOrderCached = order.inputToken.chainId !== order.outputToken.chainId
+                  const hasConfirmedBridgeData = swapAndBridgeContext?.statusResult
+
+                  // For localStorage orders: use order.outputToken immediately (no API wait, no flash)
+                  if (isLocalOrderCached) {
+                    return <CurrencyLogoPair sellToken={inputToken} buyToken={order.outputToken} tokenSize={32} />
+                  }
+
+                  // For fresh sessions: only show CurrencyLogoPair when we have confirmed bridge data
+                  if (hasConfirmedBridgeData && swapAndBridgeOverview?.targetCurrency) {
+                    return (
+                      <CurrencyLogoPair
+                        sellToken={inputToken}
+                        buyToken={swapAndBridgeOverview.targetCurrency}
+                        tokenSize={32}
+                      />
+                    )
+                  }
+
+                  // Show spinner when waiting for API data in fresh sessions (don't use CurrencyLogoPair)
+                  return (
+                    <>
+                      <TokenLogo token={inputToken} size={32} />
+                      <IconSpinner spinnerWidth={1} margin="0 0 0 -4px" size={30} />
+                    </>
+                  )
+                })()
```

Comment:
Is it possible to do this with a new component instead of an inline fn?
This could cause re-render issues if I'm not mistaken

cc @shoom3301 to confirm

Reply:
100%, it must be a component instead of IIFE
@fairlighteth